server/storage/schema/schema.go

```diff
@@ -96,12 +98,18 @@ func UnsafeDetectSchemaVersion(lg *zap.Logger, tx backend.ReadTx) (v semver.Vers
 	if vp != nil {
 		return *vp, nil
 	}
+
 	confstate := UnsafeConfStateFromBackend(lg, tx)
-	if confstate == nil {
-		return v, fmt.Errorf("missing confstate information")
-	}
 	_, term := UnsafeReadConsistentIndex(tx)
-	if term == 0 {
+	if confstate == nil && term == 0 {
+		// if both confstate and term are missing, assume it's v3.4
+		return version.V3_4, nil
+	} else if confstate == nil {
+		return v, fmt.Errorf("missing confstate information")
+	} else if len(confstate.Voters) == 0 && term == 0 {
+		// if confstate is empty and term is missing, assume it's v3.5 that was migrated from v3.4 and never started
+		return version.V3_5, nil
```

Comment:
Don't understand this case. I would prefer to avoid supporting `etcdutl migrate` from v3.4 -> v3.5.